### DISCUSSION TITLE
pfSense authentication

### DISCUSSION THREAD

**Discussion by b0bcarlson** [lldap/lldap#299] [Q&A] [answered]
I've been trying to get lldap set up for pfSense authentication but having no luck. It seems no matter what I try it fails on "Attempting to fetch Organizational Units from". Below is my latest config attempt but I have tried many combinations. Anybody have any ideas?
![image](https://user-images.githubusercontent.com/6182468/190688953-15e2a841-496a-4a31-a4e4-20d8730879ba.png)

**Comment by martadinata666** [comment]
This is my opnsense lldap, i hope will be good references ![2022-09-17-003851_937x867_scrot](https://user-images.githubusercontent.com/2433562/190699984-60ec5f9f-4d64-4630-9b09-83b157f5b7df.png)

the `extended query` : `&(objectClass=person)(memberof=cn=lldap_opnsense,ou=groups,dc=example,dc=com) `

**Comment by nitnelave** [comment]
I think your "authentication containers" is wrong. Can you try the base DN of "DC= example, DC = com" and the authentication containers of "ou = people"?

Also check your bind DN if you want "uid°auth" or "uid=readonly"

**Comment by b0bcarlson** [comment]
I was able to get it to authenticate properly (it actually may have been working before since it still gives that same message even though it works), however it doesn't seem to want to pick up the groups. I'm guessing this is something on pfSense's end since it lldap seems to be returning a proper response:
```
2022-09-17T16:01:40.993316082+00:00 DEBUG    │  ┝━ 🐛 [debug]:  | response: SearchResultEntry(LdapSearchResultEntry { dn: "uid=username,ou=people,dc=example,dc=com", attributes: [LdapPartialAttribute { atype: "memberof", vals: ["uid=admins,ou=groups,dc=example,dc=com", "uid=lldap_admin,ou=groups,dc=example,dc=com", "uid=pfsense_admin,ou=groups,dc=example,dc=com"] }] })

2022-09-17T16:01:40.993471801+00:00 DEBUG    │  ┕━ 🐛 [debug]:  | response: SearchResultDone(LdapResult { code: Success, matcheddn: "", message: "", referral: [] })

2022-09-17T16:01:40.994698197+00:00 INFO     ┕━ LDAP request [ 20.4µs | 0.01% ]
```

**Reply by nitnelave** [reply]
I think your group object class is wrong: instead of "group", try "groupOfUniqueNames"

**Reply by b0bcarlson** [reply]
That was one thing I wasn't sure of so I tried `group`, `posixGroup`, and `groupOfUniqueNames`, none of which worked. I also made sure that the groups exist locally in pfSense (I had missed that initially) which hasn't made any difference. I'm starting to think that pfSense has a bug here so I'll post there soon if I can't figure out a magic combination.

**Reply by nitnelave** [reply]
It would be interesting to see the LLDAP verbose logs, to check the query sent by pfsense and the answer. Feel free to come over on discord to troubleshoot that.

**Comment by b0bcarlson** [comment, marked as answer]
I realized I never updated. The following settings eventually did work, I believe it may have just required a restart to take effect.
![image](https://user-images.githubusercontent.com/6182468/210269264-debc4eb5-04b6-42bc-b0e1-9945d0d7c305.png)

**Reply by nitnelave** [reply]
Would you mind contributing your configuration to the example configs?

**Reply by b0bcarlson** [reply]
Absolutely!

**Comment by lordratner** [comment]
Anyone else gotten this to work?

I can get it to bind, but if fails on the test at:
"Attempting to fetch Organizational Units from "

**Reply by lordratner** [reply]
Figured it out. config example submitted
